### PR TITLE
[locator] move the config button to the tree view

### PR DESCRIPTION
**Pull request by 3nids**
screen shot 2018-05-23 at 11 44 32

as it's an improvement in UX and not really a feature, I would still merge it for 3.2 if there is no objection

### CODE REVIEW

**Comment by DelazJ** [comment]
@3nids could you add the [needs-docs] tag to allow the docs fix? Thanks

**Comment by 3nids** [comment]
@DelazJ are you sure, I don't think there is such a change. I could not find the docs for the locator, would like to give a look.

File: src/app/locator/qgslocatoroptionswidget.cpp [hunk 1]
```diff
@@ -85,6 +90,35 @@ QgsLocatorFiltersModel::QgsLocatorFiltersModel( QgsLocator *locator, QObject *pa
...
+    QWidget *w = new QWidget( parent );
+    QToolButton *bt = new QToolButton( );
+    QHBoxLayout *layout = new QHBoxLayout();
+    layout->setContentsMargins( 0, 0, 0, 0 );
+    layout->addWidget( bt );
+    w->setLayout( layout );
+
+    connect( bt, &QToolButton::clicked, this, [ = ]() {filter->openConfigWidget( bt );} );
+    //bt->setMinimumSize( 24, 24 );
+    bt->setMaximumSize( 24, 24 );
```
Comment (nyalldawson) [inline src/app/locator/qgslocatoroptionswidget.cpp]: This isn't going to work for hidpi displays - better to set using font metrics which resolve to just under 24 on low dpi and use max(fmSize, 24) to round to 24 on low dpi.

**Comment by nyalldawson** [comment]
Looks fine to me, just need the sizing fixed

**Comment by DelazJ** [comment]
@3nids You could not find docs because it's waiting for review at https://github.com/qgis/QGIS-Documentation/pull/2598/ (btw ping @rduivenvoorde)

**Comment by 3nids** [comment]
ok, nothing related to the configuration of filters here (only the main configuration dialog).

**Pull request merged**